src/builtins/fish_indent.rs

```diff
@@ -729,10 +729,40 @@ impl<'source, 'ast> PrettyPrinterState<'source, 'ast> {
             .binary_search(&brace.source_range().start())
             .is_ok()
     }
+
+    fn brace_follows_conjunction_or_pipe(&self, node: &dyn ast::Token) -> bool {
+        let brace_parent = self.traversal.parent(node.as_node());
+        let Kind::BraceStatement(_) = brace_parent.kind() else {
+            return false;
+        };
```

Comment:
probbaly this style works better, since we require exact types,
and the _parent_ bit should be obvious enough

        let Kind::BraceStatement(brace) = self.traversal.parent(node.as_node()).kind() else {
            return false;
        };